### PR TITLE
fix two unsoundness bugs in prover

### PR DESCRIPTION
**Pull request by dwightguth**
@andreistefanescu please review

One of these, the code was simply incorrect. The other (the one in Equality.isFalse) is not guaranteed to be correct during partial evaluation.

### CODE REVIEW

File: java-backend/src/main/java/org/kframework/backend/java/symbolic/Equality.java [hunk 1]
```diff
@@ -254,14 +254,16 @@ public boolean isFalse(Equality equality) {
                         rightHandSide.sort());
             } else {
                 boolean unifiable = false;
-                if (leftHandSide instanceof Variable && rightHandSide instanceof KItem) {
+                if (leftHandSide instanceof Variable && rightHandSide instanceof KItem
```
Comment (yilongli) [inline java-backend/src/main/java/org/kframework/backend/java/symbolic/Equality.java:263]: Please add some comments here with the rule that makes it incorrect during partial compilation. Most of the functionality in this class will be moved to the `SymbolicUnifier` later.

File: java-backend/src/main/java/org/kframework/backend/java/symbolic/SymbolicUnifier.java [hunk 2]
```diff
@@ -305,14 +305,12 @@ private void unifySet(BuiltinSet set, BuiltinSet otherSet) {
         Multiset<Variable> remainingVariables = Multisets.difference(variables, commonVariables);
         Multiset<Variable> otherRemainingVariables = Multisets.difference(otherVariables, commonVariables);
 
-        if (remainingElements.isEmpty()
-                && remainingPatterns.isEmpty()
```
Comment (andreistefanescu) [inline java-backend/src/main/java/org/kframework/backend/java/symbolic/SymbolicUnifier.java:309]: Why is this incorrect?

**Comment by dwightguth** [comment]
@andreistefanescu @yilongli  please re-review

File: java-backend/src/main/java/org/kframework/backend/java/symbolic/IncrementalCollector.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+                set.addAll(termSet);
+            } else {
+                setStack.push(set);
+                set = new HashSet<>();
+                setValue.accept(set, term);
+            }
+        }
+    }
+
+    private class PostVariableVisitor extends LocalVisitor {
```
Comment (andreistefanescu) [inline java-backend/src/main/java/org/kframework/backend/java/symbolic/IncrementalCollector.java]: Why is this class named PostVariableVisitor?

**Comment by andreistefanescu** [comment]
lgtm

File: java-backend/src/main/java/org/kframework/backend/java/symbolic/IncrementalCollector.java (hunk 3 above)
Comment (dwightguth) [inline java-backend/src/main/java/org/kframework/backend/java/symbolic/IncrementalCollector.java]: it was left over from before the rename of the outer class. Good catch.

**Comment by yilongli** [comment]
lgtm

**Pull request merged**